packages/flutter/lib/src/widgets/display_feature_sub_screen.dart

```diff
@@ -0,0 +1,157 @@
+  }) : super(key: key);
+
+  /// The child that will avoid the display features.
+  final Widget child;
+
+  /// The anchor point used to pick the closest area that has no display features.
+  /// `Offset(0,0)` is the top-left corner of the available screen space. For
+  /// a dual-screen device, this is the top-left corner of the left screen.
+  final Offset? anchorPoint;
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Rect> safeAreas = _safeAreasInNavigator(context);
+    final Rect safeArea = anchorPoint == null
+        ? _firstSafeArea(safeAreas, context)
+        : _closestToAnchorPoint(safeAreas, _finiteOffset(anchorPoint!));
+
+    return Align(
+      alignment: Alignment.topLeft,
+      child: Padding(
+        padding: EdgeInsets.only(left: safeArea.left, top: safeArea.top),
+        child: SizedBox(
+          width: safeArea.width,
+          height: safeArea.height,
+          child: child,
+        ),
+      ),
+    );
+  }
+
+  static Rect _firstSafeArea(List<Rect> safeAreas, BuildContext context) {
+    final TextDirection? textDirection = Directionality.maybeOf(context);
+    if (textDirection == TextDirection.rtl)
+      return _closestToAnchorPoint(safeAreas, const Offset(double.maxFinite, 0));
+    else
+      return _closestToAnchorPoint(safeAreas, Offset.zero);
+  }
+
+  static Rect _closestToAnchorPoint(List<Rect> safeAreas, Offset anchorPoint) {
+    return safeAreas.fold(safeAreas.first, (Rect previousValue, Rect element) {
+      final double previousDistance = (previousValue.center - anchorPoint).distanceSquared;
+      final double elementDistance = (element.center - anchorPoint).distanceSquared;
+      if (previousDistance < elementDistance)
+        return previousValue;
+      else
+        return element;
+    });
+  }
+
+  static List<Rect> _safeAreasInNavigator(BuildContext context) {
+    final RenderObject? renderObject = Overlay.of(context)?.context.findRenderObject();
+    Rect? navigatorBounds;
+    final Vector3? translation = renderObject?.getTransformTo(null).getTranslation();
+    if (translation != null) {
+      navigatorBounds = renderObject?.paintBounds.shift(Offset(translation.x, translation.y));
+    }
+    List<Rect> avoidBounds;
+    if (navigatorBounds == null) {
+      final Size screenSize = MediaQuery.of(context).size;
+      navigatorBounds = Rect.fromLTWH(0, 0, screenSize.width, screenSize.height);
+      avoidBounds = MediaQuery.of(context).displayFeatures
+          .map((DisplayFeature displayFeature) => displayFeature.bounds)
+          .toList();
+    } else {
+      avoidBounds = MediaQuery.of(context).displayFeatures
+          .where((DisplayFeature displayFeature) => displayFeature.bounds.overlaps(navigatorBounds!))
+          .map((DisplayFeature displayFeature) => displayFeature.bounds.shift(-navigatorBounds!.topLeft))
+          .toList();
+    }
+    return _safeAreas(Rect.fromLTWH(0, 0, navigatorBounds.width, navigatorBounds.height), avoidBounds);
```

Comment:
This seems to assume that the Navigator/Overlay is always positioned at Offset.zero?

Reply:
Please adda test case where that is not the case.